openstudiocore/src/model/FluidCoolerSingleSpeed.cpp

```diff
@@ -0,0 +1,498 @@
+    return result;
+  }
+
+  bool FluidCoolerSingleSpeed_Impl::setDesignEnteringAirWetbulbTemperature(double designEnteringAirWetbulbTemperature) {
+    bool result = setDouble(OS_FluidCooler_SingleSpeedFields::DesignEnteringAirWetbulbTemperature, designEnteringAirWetbulbTemperature);
+    return result;
+  }
+
+  bool FluidCoolerSingleSpeed_Impl::setDesignWaterFlowRate(boost::optional<double> designWaterFlowRate) {
+    bool result(false);
+    if (designWaterFlowRate) {
+      result = setDouble(OS_FluidCooler_SingleSpeedFields::DesignWaterFlowRate, designWaterFlowRate.get());
+    }
+    return result;
+  }
+
+  void FluidCoolerSingleSpeed_Impl::autosizeDesignWaterFlowRate() {
+    bool result = setString(OS_FluidCooler_SingleSpeedFields::DesignWaterFlowRate, "autosize");
+    OS_ASSERT(result);
+  }
+
+  bool FluidCoolerSingleSpeed_Impl::setDesignAirFlowRate(boost::optional<double> designAirFlowRate) {
+    bool result(false);
+    if (designAirFlowRate) {
+      result = setDouble(OS_FluidCooler_SingleSpeedFields::DesignAirFlowRate, designAirFlowRate.get());
+    }
+    return result;
+  }
+
+  void FluidCoolerSingleSpeed_Impl::autosizeDesignAirFlowRate() {
+    bool result = setString(OS_FluidCooler_SingleSpeedFields::DesignAirFlowRate, "autosize");
+    OS_ASSERT(result);
+  }
+
+  bool FluidCoolerSingleSpeed_Impl::setDesignAirFlowRateFanPower(boost::optional<double> designAirFlowRateFanPower) {
+    bool result(false);
+    if (designAirFlowRateFanPower) {
+      result = setDouble(OS_FluidCooler_SingleSpeedFields::DesignAirFlowRateFanPower, designAirFlowRateFanPower.get());
+    }
+    return result;
+  }
+
+  void FluidCoolerSingleSpeed_Impl::autosizeDesignAirFlowRateFanPower() {
+    bool result = setString(OS_FluidCooler_SingleSpeedFields::DesignAirFlowRateFanPower, "autosize");
+    OS_ASSERT(result);
+  }
+
+  bool FluidCoolerSingleSpeed_Impl::setOutdoorAirInletNode(const boost::optional<Node>& node) {
+    bool result(false);
+    if (node) {
+      result = setPointer(OS_FluidCooler_SingleSpeedFields::OutdoorAirInletNode, node.get().handle());
+    }
+    else {
+      resetOutdoorAirInletNode();
+      result = true;
+    }
+    return result;
+  }
+
+  void FluidCoolerSingleSpeed_Impl::resetOutdoorAirInletNode() {
+    bool result = setString(OS_FluidCooler_SingleSpeedFields::OutdoorAirInletNode, "");
+    OS_ASSERT(result);
+  }
+
+} // detail
+
+FluidCoolerSingleSpeed::FluidCoolerSingleSpeed(const Model& model)
+  : StraightComponent(FluidCoolerSingleSpeed::iddObjectType(),model)
+{
+  OS_ASSERT(getImpl<detail::FluidCoolerSingleSpeed_Impl>());
+
+  // TODO: Appropriately handle the following required object-list fields.
+  bool ok = true;
+   //ok = setHandle();
+  OS_ASSERT(ok);
+   ok = setPerformanceInputMethod("NominalCapacity"); // UFactorTimesAreaAndDesignWaterFlowRate, NominalCapacity
+  OS_ASSERT(ok);
+  ok = setDesignAirFlowRateUfactorTimesAreaValue(0.1);
```

Comment:
These don't look like very good defaults to me.  Use E+ idd default or better yet us material from an example file.